Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+			m_DoorMoveSound.Stop();
+			if (!m_ResumeAfterStop) {
+				m_DoorDirectionChangeSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_DoorMoveTime - m_DoorMoveTimer.GetElapsedSimTimeMS());
+			} else {
+				m_DoorMoveStartSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_ChangedDirectionAfterStop ? m_DoorMoveTime - m_DoorMoveStopTime : m_DoorMoveStopTime);
+				m_ChangedDirectionAfterStop = false;
+				m_ResumeAfterStop = false;
+			}
+		} else if (m_DoorState == STOPPED) {
+			if (m_DoorMaterialDrawn) { EraseDoorMaterial(); }
+			m_ResumeAfterStop = true;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::Update() {
+		Actor::Update();
+
+		if (m_Door && m_Door->IsAttached()) {
+			if (m_DoorState != STOPPED && m_SensorTimer.IsPastSimMS(m_SensorInterval)) { UpdateSensors(); }
+			UpdateDoorAttachable();
+		}
+
+		if (m_Door && !ApplyAttachableForces(m_Door)) {
+			// Remove the material representation if the door fell off
+			EraseDoorMaterial();
+			// Start the spinning out of control animation for the motor, start it slow
+			m_SpriteAnimDuration *= 4;
+			m_Door = 0;
+		}
+
+		if (m_SpriteAnimMode == LOOPWHENOPENCLOSE && m_FrameCount > 1 && (m_DoorState == OPENING || m_DoorState == CLOSING) && m_SpriteAnimTimer.IsPastSimMS(m_SpriteAnimDuration)) {
+			m_Frame = (m_Frame + 1) % m_FrameCount;
+			m_SpriteAnimTimer.Reset();
+		}
+
+		// Lose health when door is lost, spinning out of control until grinds to halt
+		if (!m_Door && m_Status != DYING && m_Status != DEAD) {
+			m_SpriteAnimMode = ALWAYSLOOP;
+			m_SpriteAnimDuration = LERP(0, m_MaxHealth, 10, m_InitialSpriteAnimDuration, m_Health);
+
+			// TODO: Figure out how to pitch this correctly so it goes up as SpriteAnimDuration goes down.
+			/*
+			if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+			g_AudioMan.SetSoundPitch(&m_DoorMoveSound, LERP(0, m_InitialSpriteAnimDuration, 0, 5, m_SpriteAnimDuration));
+			*/
+
+			m_Health -= 0.4F;
+		}
+		if (m_Status == DEAD) { GibThis(Vector(), 50); }
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::UpdateSensors() {		
+		const Actor *foundActor = 0;
+		bool anySensorInput = false;
+
+		for (ADSensor &sensor : m_Sensors) {
+			foundActor = sensor.SenseActor(m_Pos, m_Rotation, m_HFlipped, m_MOID);
+			if (foundActor && foundActor->IsControllable()) {
+				anySensorInput = true;
+				// Look for enemy actors first and break the loop so we don't accidentally open the door for a friendly actor is enemies are nearby.
+				if (foundActor->GetTeam() != m_Team) {
+					CloseDoor();
+					break;
+				} else if (foundActor->GetTeam() == m_Team) {					
+					OpenDoor();
+				}
+			}
+		}
+		// If no sensor input, return to default state if it's time. This won't override any actions that the sensors started
```

Comment:
No need for this comment, the only thing of note is about overriding, and that's not really very helpful imo, especially since your method comment already explains it.